specs/WebResourceRequested-CustomScheme.md

```diff
@@ -0,0 +1,296 @@
+  // The URIs of registered custom schemes will be treated similar to http URIs for their origins.
+  // They will have tuple origins for URIs with host and opaque origins for
+  // URIs without host as specified in [7.5 Origin - HTML Living Standard](https://html.spec.whatwg.org/multipage/origin.html)
+  // Example:
+  // custom-scheme-with-host://hostname/path/to/resource has origin of custom-scheme-with-host://hostname
+  // custom-scheme-without-host:path/to/resource has origin of custom-scheme-without-host:path/to/resource
+  // For WebResourceRequested event, the cases of request URIs and filter URIs with custom schemes
+  // will be normalized according to generic URI syntax rules. Any non-ASCII characters will be preserved.
+  // The registered custom schemes also participate in [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS?msclkid=79c64f88a64c11ec8862c1ba8d05164c) and adheres to [CSP](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP?msclkid=870484bca64c11ecbe57b10ab5f74c35). The app needs to set the appropriate access headers in its WebResourceRequested event handler to allow CORS requests.
+
+  // The name of the custom scheme to register.
+  [propget] HRESULT SchemeName([out, retval] LPCWSTR* schemeName);
+  [propput] HRESULT SchemeName([in] LPCWSTR value);
+
+  // Whether the scheme will be treated as a [Secure Context](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts) like https.
+  [propget] HRESULT IsSecure([out, retval] BOOL* isSecure);
+  // Set if the scheme will be treated as a Secure Context.
+  [propput] HRESULT IsSecure([in] BOOL value);
+
+  // Array of origins that are allowed to issue requests with the custom scheme.
+  // Except origins with this same custom scheme, which are always allowed,
+  // the origin of any request (requests that have the 
+  // [Origin header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin?msclkid=f7147fe3a64711ecbcde6eb3114a9946)) to the custom scheme URL
+  // needs to be in this list. No-origin requests are requests that do not have an Origin header,
+  // such as link navigations, embedded images and are always allowed.
+  // Note that cross-origin restrictions still apply. 
+  // If the list is empty, no cross-origin request to this scheme is allowed.
+  // Origins are specified as a string in the format of scheme://host:port.
+  // The origins are string pattern matched with `*` (matches 0 or more characters)
+  // and `?` (matches 0 or 1 character) wildcards just like the URI matching in the
+  // [AddWebResourceRequestedFilter API](https://docs.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2.addwebresourcerequestedfilter?msclkid=33bd40dda64d11ec9390310f4f42d68a&view=webview2-dotnet-1.0.1150.38).
+  // For example, "http://*.example.com:80".
+  // The array of strings and the array here must be deallocated with CoTaskMemFree if
```

Comment:
Should drop the 'if using the ...' text. Anyone implementing this interface must follow the same rules and produce an array that should be deallocated with CoTaskMemFree.